TranslatorReasonerAPI.yaml

```diff
@@ -784,10 +796,69 @@ components:
           items:
             $ref: '#/components/schemas/Attribute'
           nullable: true
+        qualifiers:
+          type: array
+          description: >-
+            A list of qualifiers that together provide additional detailed
+            nuance to the assertion beyond what the subject-predicate-object
+            can provide.
+          items:
+            $ref: '#/components/schemas/Qualifier'
+          nullable: true
       additionalProperties: false
       required:
         - subject
         - object
+    Qualifier:
+      additionalProperties: false
+      description: >-
+        An additional nuance attached to an assertion
+      type: object
+      properties:
+        qualifier_type_id:
+          type: string
+          description: >-
+            The category of the qualifier, drawn from a hierarchy of qualifier
+            slots in the Biolink model (e.g. subject_aspect, subject_direction,
+            object_aspect, object_direction, etc).
+          example: subject_aspect
+          nullable: false
+        value:
+          type: string
+          description: >-
+            The value associated with the type of the qualifier, drawn from
+            a set of controlled values by the type as specified in
+            the Biolink model (e.g. 'decreased' or 'increased' for the type
+            subject_direction, etc).
+          example: decreased
+          nullable: false
+      required:
+        - type
+        - value
+    QualifierSet:
+      additionalProperties: false
+      description: >-
+        A set of Qualifiers that act together to add nuance to an assertion.
+      properties:
+        qualifiers:
+          type: array
+          description: >-
+            A list of qualifiers that together provide additional detailed
+            nuance to the assertion beyond what the subject-predicate-object
+            can provide.
+          items:
+            $ref: '#/components/schemas/Qualifier'
+          nullable: false
+        qualified_predicate:
+          $ref: '#/components/schemas/BiolinkPredicate'
+          description: >-
+            The predicate that is to be qualified in cases where there
+            are multiple predicates in a query and there thus might be
+            ambiguity.
```

Comment:
In this initial draft of an object to support Qualifier-based query constraints, I would also include a boolean 'not' property, as exists for the QueryConstraint object used for Attribute-based constraints . Key for the basic functionality of excluding specific qualifiers from a result.

Reply:
This suggestion was motivated by an example query Valdo provided that asks "What compounds increase the activity or phosphorylation of PPARA but **not** its abundance".  But as you point out, negation/exclusion are tricky and perhaps it is best to leave this out for now, until we look into it more closely.